### ISSUE TITLE
integralClosure(Ideal) incorrectly returns unit ideal 

### ISSUE THREAD

**Issue by devlinmallory** [Macaulay2/M2#846]
I believe that integralClosure(Ideal) is incorrectly returning the zero ideal in versions 1.11 and 1.12. The following behavior is seen in version 1.10 (as checked on http://habanero.math.cornell.edu:3690/):
```
Macaulay2, version 1.10
with packages: ConwayPolynomials, Elimination, IntegralClosure, InverseSystems, LLLBases,
               PrimaryDecomposition, ReesAlgebra, TangentCone

i1 : R=QQ[x,y]

o1 = R

o1 : PolynomialRing

i2 : I=ideal(x^2,y^2)

             2   2
o2 = ideal (x , y )

o2 : Ideal of R

i3 : integralClosure I

             2        2
o3 = ideal (y , x*y, x )

o3 : Ideal of R
```
The same occurs in version 1.8.2 running on my machine. 

In contrast, in 1.11 and 1.12 (as tested on my machine), the following occurs:
```
Macaulay2, version 1.12
with packages: ConwayPolynomials, Elimination, IntegralClosure, InverseSystems,
               LLLBases, PrimaryDecomposition, ReesAlgebra, TangentCone

i1 : R=QQ[x,y]

o1 = R

o1 : PolynomialRing

i2 : I=ideal(x^2,y^2)

             2   2
o2 = ideal (x , y )

o2 : Ideal of R

i3 : integralClosure I

o3 = ideal ()

o3 : Ideal of R
```
Any ideas? 

Update 2/26: Just wanted to flag that this appears to still be broken in version 1.13- I think Dan's comment below is likely the resulting issue. Additionally, just wanted to note this breaks other methods like minimalReduction (unsurprisingly), in case others experience issues with those functions.

**Comment by DanGrayson** [comment]
I think it might derive from this change:
```
Macaulay2, version 1.10
with packages: ConwayPolynomials, Elimination, IntegralClosure, LLLBases, PrimaryDecomposition,
               ReesAlgebra, TangentCone

i1 : R=QQ[x,y]; reesAlgebra ideal(x^2,y^2)

      R[w , w ]
         0   1
o2 = -----------
      2      2
     y w  - x w
        0      1

o2 : QuotientRing

=============================================================================

Macaulay2, version 1.12.0.1
with packages: ConwayPolynomials, Elimination, IntegralClosure, InverseSystems, LLLBases,
               PrimaryDecomposition, ReesAlgebra, TangentCone

i1 : R=QQ[x,y]; reesAlgebra ideal(x^2,y^2)

      R[w , w ]
         0   1
o2 = -----------
      2      2
     x w  - y w
        0      1

o2 : QuotientRing
```
and an assumption about such results implicit in the integral closure package.

**Comment by mikestillman** [comment]
There has been a thread about this on the google group.  It needs to be fixed, somehow I overlooked this.

Here is an example too (still fails in M2 1.14).

```
QQ[x,y]
integralClosure ideal (x^3,x*y,y^2)
```
@eisenbud , did you write the integral closure of ideal code?

**Comment by eisenbud** [comment]
Probably so. The idea was to form the Rees algebra and take the integral closure of that. The Rees algebra code has probably changed in the meantime.

(I have no idea how to reply on github, by the way.)

David
--
David Eisenbud
Director, Mathematical Sciences Research Institute; and
Professor of Mathematics, University of California, Berkeley
www.msri.org/~de

**Comment by pzinn** [comment]
I may be responsible for this fiasco via
https://github.com/Macaulay2/M2/commit/d64ea286a84498ee6d96c6cb5856dc78b42e5b56#diff-053c2b9a6cb2fea82688e37e938195cf
I haven't had time to look, but I'm pretty sure my pull changed the indexing of the variables (which I naively assumed would be irrelevant). the pull can be easily reverted, though the bug it fixed will reappear.

**Comment by mikestillman** [comment]
Let me first look, and see if we can fix it in a more robust manner.

**Comment by DanGrayson** [comment]
We should have had more test cases.

(David, replying to the email from github puts your reply on github.)

**Comment by eisenbud** [comment]
I'll work on this at the M2 meeting in Germany...

David
--
David Eisenbud
Director, Mathematical Sciences Research Institute; and
Professor of Mathematics,University of California, Berkeley
www.msri.org/~de

**Comment by mikestillman** [comment]
Fixed via commit f1cea48.
